src/nft/components/mint/NFTMintButton.tsx

```diff
@@ -0,0 +1,100 @@
+import { useCallback, useMemo } from 'react';
+import { useAccount, useChainId } from 'wagmi';
+import { cn } from '../../../styles/theme';
+import {
+  Transaction,
+  TransactionButton,
+  type LifecycleStatus as TransactionLifecycleStatus,
+  TransactionSponsor,
+  TransactionStatus,
+  TransactionStatusAction,
+  TransactionStatusLabel,
+} from '../../../transaction';
+import { ConnectWallet } from '../../../wallet';
+import { useNFTLifecycleContext } from '../NFTLifecycleProvider';
+import { useNFTContext } from '../NFTProvider';
+
+type NFTMintButtonReact = {
+  className?: string;
+  label?: string;
+};
+
+export function NFTMintButton({
+  className,
+  label = 'Mint',
+}: NFTMintButtonReact) {
+  const chainId = useChainId();
+  const { address } = useAccount();
+  const {
+    contractAddress,
+    tokenId,
+    isEligibleToMint,
+    buildMintTransaction,
+    quantity,
+  } = useNFTContext();
+  const { updateLifecycleStatus } = useNFTLifecycleContext();
+
+  const handleOnStatus = useCallback(
+    (transactionStatus: TransactionLifecycleStatus) => {
+      if (transactionStatus.statusName === 'transactionPending') {
+        updateLifecycleStatus({ statusName: 'transactionPending' });
+      }
+
+      if (
+        transactionStatus.statusName === 'transactionLegacyExecuted' ||
+        transactionStatus.statusName === 'success' ||
+        transactionStatus.statusName === 'error'
+      ) {
+        updateLifecycleStatus(transactionStatus);
+      }
+    },
+    [updateLifecycleStatus],
+  );
+
+  const transactionButtonLabel = useMemo(() => {
+    if (isEligibleToMint === false) {
+      return 'Mint ended';
+    }
+
+    return label;
+  }, [isEligibleToMint, label]);
+
+  if (!buildMintTransaction) {
+    return null;
+  }
+
+  if (!address) {
+    return (
+      <div className={cn('py-2', className)}>
+        <ConnectWallet className="w-full" />
+      </div>
+    );
+  }
```

Comment:
we should align our components to **_always_** render, and only prompt for wallet connection in response to an action (e.g. minting the NFT)

Reply:
agreed you can see my implementation with SwapButton here https://github.com/coinbase/onchainkit/blob/main/src/swap/components/SwapButton.tsx

Reply:
I'm missing the difference

Reply:
this is the same implementation as our swap button https://github.com/coinbase/onchainkit/blob/main/src/swap/components/SwapButton.tsx

Reply:
are you suggesting we change the `onClick` handler to be connect wallet if address isn't connected ?